### DISCUSSION TITLE
Adjust DataTable scrollHeight attribute within the window resize event handler

### DISCUSSION THREAD

**Discussion by manuelpoelzl** [primefaces/community#95] [PrimeReact] [answered]
Hi everyone,

Within my application I do have a DataTable which has set the scrollHeight property to this:
```jsx
scrollHeight={(window.innerHeight-290).toString()+"px"}}
```
This works perfectly fine, however i want to achieve the same behavior when I resize the window.
I tried to set the scrollHeight attribute in the event listener but saw that the scrollHeight attribute of the `getTable()` return value can not be changed.

I'm using the table ref which i defined like this:
```jsx
const tableRef=useRef<DataTable<any>>(null);
```

But I don't see an option here to define the desired attribute, so i wonder if there is another way to achieve this?

**Comment by melloware** [comment, marked as answer]
Can't you change this to a `useState` hook so it rerenders on change?

```js
{(window.innerHeight-290).toString()+"px"}}
```

**Reply by manuelpoelzl** [reply]
Well, you're right.
Forgot for a moment how React works it seems to me 😅

Sorry for the inconvenience and thanks for answering!!

**Reply by melloware** [reply]
No problem!!!

**Reply by manuelpoelzl** [reply]
Hi!
Do you maybe know some sort of workaround when using the `virtualScrollerOptions`?
The resize works perfectly fine when i shrink the window, the size which is set as state is applied correctly:
![Screenshot_9](https://github.com/primefaces/community/assets/46539758/03e97010-088c-4f50-9ca1-1976bb524a96)

However, when i enlarge the window again, the table seems kinda cut off:
![Screenshot_10](https://github.com/primefaces/community/assets/46539758/220b221d-61e0-4e02-b003-9e8f75fd4295)

I doubt anybody will ever do it, but when I shrink the window to some really small size and enlarge it again, the whole content seems to be gone:
![Screenshot_12](https://github.com/primefaces/community/assets/46539758/9280dc52-e5a7-415e-b198-7cdeee726d11)

I've tried to use some of the `virtualScrollerOptions` such as `scrollHeight` and `autoSize` but had no luck with it.

**Reply by melloware** [reply]
I think your issue is this open bug: https://github.com/primefaces/primereact/issues/4413

**Reply by manuelpoelzl** [reply]
Thanks for the hint!
That is indeed the same problem.
